### ISSUE TITLE
Convert model to ONNX format

### ISSUE THREAD

**Issue by manhminno** [davidsandberg/facenet#1185]
Hello,
I'm trying to convert .pb model to Onnx format. But there was an error.
#### Command: 
``` 
python -m tf2onnx.convert --input 20180402-114759.pb--inputs input:0,phase_train:0 --outputs embeddings:0 --output model_2311.onnx --verbose
```
#### Error:
_ValueError: NodeDef expected inputs '' do not match 1 inputs specified; Op<name=Const; signature= -> output:dtype; attr=value:tensor; attr=dtype:type>; NodeDef: {{node InceptionResnetV1/Conv2d_1a_3x3/BatchNorm/cond/Const}} = Const\[dtype=DT_FLOAT, value=Tensor<type: float shape: [0] values: >\](InceptionResnetV1/Conv2d_1a_3x3/BatchNorm/cond/Switch:1)_

Thank!

**Comment by Ram-Godavarthi** [comment]
Did you able to get onnx model ?
Could you share the command to get it.

**Comment by juransaber** [comment]
import tensorflow as tf
import numpy as np
import tf2onnx;
import onnxruntime as ort  
import os;

# 加载冻结的 .pb 文件（需从官方仓库下载）
with tf.io.gfile.GFile('20180408-102900/20180408-102900.pb', 'rb') as f:
    graph_def = tf.compat.v1.GraphDef()
    graph_def.ParseFromString(f.read())

output_path = "model.onnx" 

for node in graph_def.node:
    if node.op == "Placeholder":
        print(f"输入节点: {node.name}")
    if "embeddings" in node.name:
        print(f"输出节点: {node.name}")

x = np.array([5.], dtype=np.float32)
shape_override = {
    "phase_train:0": [],  # 用 None 表示动态batch
    "batch_size:0": [],  # 用 None 表示动态batch
    "batch_join:0": [1, 160, 160, 3],  # 用 None 表示动态batch
    "batch_join:1": [],  # 用 None 表示动态batch
    "embeddings:0": [1, 512]           # 输出维度同理
}
model_proto, _ = tf2onnx.convert.from_graph_def(graph_def, 
    input_names=["batch_size:0", "phase_train:0"], 
    output_names=["embeddings:0"],
    shape_override=shape_override,
    opset=12, output_path=output_path)
output_names = [n.name for n in model_proto.graph.output]

print("output_names:", output_names);

**Comment by juransaber** [comment]
not sure, but no errors

**Comment by Shun7shun** [comment]
> not sure, but no errors



> import tensorflow as tf import numpy as np import tf2onnx; import onnxruntime as ort import os;
> 
> # 加载冻结的 .pb 文件（需从官方仓库下载）
> with tf.io.gfile.GFile('20180408-102900/20180408-102900.pb', 'rb') as f: graph_def = tf.compat.v1.GraphDef() graph_def.ParseFromString(f.read())
> 
> output_path = "model.onnx"
> 
> for node in graph_def.node: if node.op == "Placeholder": print(f"输入节点: {node.name}") if "embeddings" in node.name: print(f"输出节点: {node.name}")
> 
> x = np.array([5.], dtype=np.float32) shape_override = { "phase_train:0": [], # 用 None 表示动态batch "batch_size:0": [], # 用 None 表示动态batch "batch_join:0": [1, 160, 160, 3], # 用 None 表示动态batch "batch_join:1": [], # 用 None 表示动态batch "embeddings:0": [1, 512] # 输出维度同理 } model_proto, _ = tf2onnx.convert.from_graph_def(graph_def, input_names=["batch_size:0", "phase_train:0"], output_names=["embeddings:0"], shape_override=shape_override, opset=12, output_path=output_path) output_names = [n.name for n in model_proto.graph.output]
> 
> print("output_names:", output_names);

大佬您好，基于这个转出来的ONNX，可以使用ONNXRuntime推理[validate_on_lfw.py](https://github.com/davidsandberg/facenet/blob/master/src/validate_on_lfw.py)这个吗？
